src/components/Detailpage/Detailpage.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { ReactComponent as Gluten } from '../../images/icons/gluten.svg'
+import { ReactComponent as Soy } from '../../images/icons/soy.svg'
+import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
+import { useHistory, useParams } from 'react-router-dom'
+
+export default function Detailpage({ category }) {
+  let { id } = useParams()
+  useEffect(() => {
+    fetchDetail()
+  }, [])
+
+  const [detail, setDetail] = useState({
+    allergies: [],
+  })
+
+  const fetchDetail = async () => {
+    const fetchDetail = await fetch(`http://localhost:4000/${category}/${id}`)
+    const detail = await fetchDetail.json()
+    setDetail(detail)
+  }
+
+  let history = useHistory()
+
+  function handleClick() {
+    history.goBack()
+  }
+
+  return (
+    <>
+      <ImageSection>
+        <ArrowStyled onClick={handleClick} />
+        <ProductName>{detail.title}</ProductName>
+        <ProductImage src={detail.image} />
+      </ImageSection>
+      <ProductInformation>
+        <Headline>Ingredients</Headline>
+        <IngredientsList>{detail.ingredients}</IngredientsList>
+        {detail.allergies.length ? (
+          <Headline>Allergies</Headline>
+        ) : (
+          <Headline></Headline>
+        )}
+        {detail.allergies.map((allergy, index) => (
+          <AllergyContainer key={index}>
+            {allergy === 'Gluten' ? <Gluten /> : <Soy />}
+            <Allergies>{allergy}</Allergies>
+          </AllergyContainer>
+        ))}
+      </ProductInformation>
+    </>
+  )
+}
+
+const ImageSection = styled.section`
+  display: flex;
+  background: linear-gradient(
+    45deg,
+    rgba(75, 219, 164, 1) 0%,
+    rgba(75, 219, 128, 1) 100%
+  );
+  border-radius: 0 0 20px 20px;
+  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.2);
+  width: 100%;
+  grid-column: 1/3;
+`
+const ProductName = styled.h2`
+  font-size: 22px;
+  font-weight: 300;
+  line-height: 1.5;
+  color: white;
+  margin: 100px 20px 30px 30px;
+  width: 40vw;
+`
+
+const ProductImage = styled.img`
+  height: 180px;
+  text-align: center;
+  -webkit-filter: drop-shadow(0 2px 6px rgba(0, 0, 0, 0.2));
```

Comment:
You might not need this as this create-react-app should have `autoprefixer` included.